File: backend/core/src/modules/relationships/relationships.controller.ts
import { Controller, UseGuards } from '@nestjs/common';
import { ProjectScopedCrudController } from '../../common/access/project-scoped-crud.controller';
import { RelationshipsService } from './relationships.service';
import { Relationship } from '../../entities';
import { JwtAuthGuard } from '../auth/jwt-auth.guard';

@UseGuards(JwtAuthGuard)
@Controller('relationships')
export class RelationshipsController extends ProjectScopedCrudController<Relationship> {
  constructor(service: RelationshipsService) {
    super(service);
  }
}

File: backend/core/src/main/java/store/pjcloud/cowrite/core/controller/RelationshipsController.java
package store.pjcloud.cowrite.core.controller;

import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import store.pjcloud.cowrite.core.common.ProjectScopedCrudController;
import store.pjcloud.cowrite.core.entity.Relationship;
import store.pjcloud.cowrite.core.service.RelationshipsService;

@RestController
@RequestMapping("/relationships")
public class RelationshipsController extends ProjectScopedCrudController<Relationship> {
    public RelationshipsController(RelationshipsService service) {
        super(service);
    }
}
